src/main/java/com/google/devtools/build/lib/skyframe/ActionMetadataHandler.java

```diff
@@ -727,7 +727,8 @@ private static FileArtifactValue fileArtifactValueFromStat(
   }
 
   private void setPathPermissionsIfFile(Path path) throws IOException {
-    if (path.isFile(Symlinks.NOFOLLOW)) {
+    FileStatus stat = path.stat(Symlinks.NOFOLLOW);
```

Comment:
Should this use `statIfFound`? All of the other stats in this file use that to tolerate a nonexistent file.

Reply:
True, it should to preserve the original semantics. I pushed a commit to do this. I am a bit surprised that no test failed due to this change - do all missing files at this point result in a `missing outputs` error down the line anyway?